modules/preloader/include/mistAPI.js

```diff
@@ -159,20 +157,23 @@ module.exports = () => {
                 const filteredId = prefix + filterId(id);
                 queue.push({ action: 'selectMenu', id: filteredId });
 
-                for (var e in this.entries) {
+                for (let e in this.entries) {
                     if ({}.hasOwnProperty.call(this.entries, e)) {
                         this.entries[e].selected = (e === filteredId);
                     }
                 }
+                executeCallback(null, filteredId);
```

Comment:
Not sure, if calling `select(id)`, should also call the callback for that menu entry..
this api is mainly for updating the menu in mist, not for calling actions in the developers dapp. For this he will have to program his own logic.

The idea would be also that the mist api is not available sometimes, so you wouldn't depend core logic of your app on it. You would want to do it like this:

```js
'click .mySuperButton': function(){

    MyApp.goToPath('/super');

    // also select entry, if mist is available
    if(window.mist) {
        mist.select('superButton');
    }
}
```